src/videojs-hls.js

```diff
@@ -74,6 +74,38 @@ videojs.Hls.prototype.src = function(src) {
   this.segmentBuffer_ = [];
   this.segmentParser_ = new videojs.Hls.SegmentParser();
 
+  // if the stream contains ID3 metadata, expose that as a metadata
+  // text track
+  (function() {
+    var
+      metadataStream = tech.segmentParser_.metadataStream,
+      textTrack;
+
+    metadataStream.on('data', function(metadata) {
+      var i, frame, time, hexDigit;
+
+      // create the metadata track if this is the first ID3 tag we've
+      // seen
+      if (!textTrack) {
+        textTrack = tech.player().addTextTrack('metadata', 'Timed Metadata');
+
+        // build the dispatch type from the stream descriptor
+        // https://html.spec.whatwg.org/multipage/embedded-content.html#steps-to-expose-a-media-resource-specific-text-track
+        textTrack.inBandMetadataTrackDispatchType = videojs.Hls.SegmentParser.STREAM_TYPES.metadata.toString(16);
+        for (i = 0; i < metadataStream.descriptor.length; i++) {
+          hexDigit = ('00' + metadataStream.descriptor[i].toString(16)).slice(-2);
+          textTrack.inBandMetadataTrackDispatchType += hexDigit;
+        }
+      }
+
+      for (i = 0; i < metadata.frames.length; i++) {
+        frame = metadata.frames[i];
+        time = metadata.pts / 1000;
+        textTrack.addCue(new window.VTTCue(time, time, frame.value || frame.url));
```

Comment:
I was going to mention that on iOS6 or 7, we only have TextTrackCue and not VTTCue but then I realized this doesn't apply to iOS.